src/atoms/RadioButton/RadioButton.native.js

```diff
@@ -0,0 +1,371 @@
+            height: '10px',
+            borderRadius: '5px',
+            borderWidth: '1px',
+          };
+        default:
+          return {
+            width: '12px',
+            height: '12px',
+            borderRadius: '6px',
+            borderWidth: '1.2px',
+          };
+      }
+    },
+  },
+  dot: {
+    dimensions(size) {
+      switch (size) {
+        case 'large':
+          return {
+            width: '7px',
+            height: '7px',
+            borderRadius: '3.5px',
+          };
+        case 'medium':
+          return {
+            width: '6px',
+            height: '6px',
+            borderRadius: '3px',
+          };
+        case 'small':
+          return {
+            width: '5px',
+            height: '5px',
+            borderRadius: '2.5px',
+          };
+        default:
+          return {
+            width: '6px',
+            height: '6px',
+            borderRadius: '3px',
+          };
+      }
+    },
+  },
+};
+
+const Dot = styled(View)(
+  (props) =>
+    `
+    border-radius: ${props.borderRadius};
+    background-color: ${props.backgroundColor};
+  `,
+);
+
+const Circle = styled(View)(
+  (props) =>
+    `
+    border-radius: ${props.borderRadius}; 
+    border-color: ${props.color};
+    border-width: ${props.borderWidth}; 
+  `,
+);
+
+const isChecked = (context, currentValue) => {
+  return context && isDefined(context.value) && context.value === currentValue;
+};
+
+const RadioButtonContext = React.createContext(null);
+
+const Radio = ({ value, onValueChange, defaultValue, children }) => {
+  const [selected, setSelected] = useState(value || defaultValue);
+
+  const onChange = useCallback(
+    (newValue) => {
+      if (!value) {
+        setSelected(newValue);
+      }
+      if (isDefined(onValueChange)) onValueChange(newValue);
+    },
+    [onValueChange, value],
+  );
+
+  return (
+    <RadioButtonContext.Provider value={{ value: selected, onValueChange: onChange }}>
+      {children}
+    </RadioButtonContext.Provider>
+  );
+};
+
+const RadioButton = ({
+  size,
+  value,
+  disabled,
+  onClick,
+  title,
+  helpText,
+  errorText,
+  variantColor,
+  testID,
+}) => {
+  let titleTextColor = 'shade.980';
+  let helpTextColor = 'shade.950';
+  const [underlayColor, setUnderlayColor] = useState('transparent');
+  const context = useContext(RadioButtonContext);
+  const theme = useContext(ThemeContext);
+  const checked = isChecked(context, value);
+
+  const radioColor = styles.radio.color({ theme, disabled, checked, variantColor });
+
+  const onPressIn = useCallback(() => {
+    let colorKey = 'tone.940';
+    if (checked) {
+      colorKey = `${variantColor || 'primary'}.930`;
+    }
+    const newUnderlayColor = getColor(theme, colorKey);
+    setUnderlayColor(newUnderlayColor);
+  }, [checked, theme, variantColor]);
+
+  const onPressOut = useCallback(() => {
+    setUnderlayColor('transparent');
+  }, []);
+
+  const onPress = useCallback(() => {
```

Comment:
Can we just use `onPressIn` and `onPressOut`? `onPress` seems to be redundant.  `onPressIn` and `onPressOut` together satisfies the needs.

Reply:
@deorukhkarchaitanya @kamleshchandnani

Reply:
:O 

Interesting observation @shubhnik ! Let me get back to you on this

Reply:
Okay so,

When you click on a button the call sequence is

1) onPressIn
2) onPressOut
3) onPress

In this case all 3 are called so we can replace onPress with onPressOut

**But**,

When you hold down on a button and swipe your finger away from it, this is the sequence of calls


1) onPressIn
2) onPressOut

onPress is **never** called. This indicates that the user pressed the button, then decided to not take that action by moving their finger away, so only onPressOut is called where we'd ideally like to disable our "active" state without calling the onPress action


Check out to this branch and see the logs to understand the behaviour: ```test/onPress```